Mathlib/Topology/ContinuousFunction/CFC.lean

```diff
@@ -0,0 +1,1111 @@
+spectrum is unique. This utilizes the Stone-Weierstrass theorem. -/
+instance [T2Space A] {a : A} [CompactSpace (spectrum 𝕜 a)] : Subsingleton (CFCCoreClass 𝕜 a) :=
+  Subsingleton.intro fun h₁ h₂ => h₁.ext h₂ <|
+    ContinuousMap.starAlgHom_ext_map_X h₁.hom_continuous h₂.hom_continuous <|
+      h₁.hom_map_X.trans h₂.hom_map_X.symm
+
+variable [TopologicalRing A] [ContinuousStar A] (a : A) [CompactSpace (spectrum 𝕜 a)]
+    [CFCCoreClass 𝕜 a]
+
+variable (𝕜)
+
+/-- The range of the continuous functional calculus is contained in the `elementralStarAlgebra`
+generated by the element. -/
+theorem cfc₁_range_le : (cfc₁ 𝕜 a).range ≤ elementalStarAlgebra 𝕜 a := by
+  rw [StarAlgHom.range_eq_map_top, ← polynomialFunctions.starClosure_topologicalClosure]
+  refine (StarSubalgebra.map_topologicalClosure_le _ _ (cfc₁_continuous 𝕜 a)).trans <|
+    StarSubalgebra.topologicalClosure_mono ?_
+  rw [polynomialFunctions.starClosure_eq_adjoin_X, StarAlgHom.map_adjoin]
+  refine StarSubalgebra.adjoin_le ?_
+  simpa only [Set.image_singleton, Set.singleton_subset_iff, toContinuousMapOnAlgHom_apply,
+    cfc₁_map_X] using StarSubalgebra.self_mem_adjoin_singleton 𝕜 a
+
+/-- The range of the continuous functional calculus is contained in the `elementralStarAlgebra`
+generated by the element. -/
+theorem cfc₂_range_le : (cfc₂ 𝕜 a).range ≤ elementalStarAlgebra 𝕜 a := by
+  rintro _ ⟨f, rfl⟩
+  exact cfc₁_range_le 𝕜 a ⟨_, rfl⟩
+
+variable {𝕜}
+
+/-- The range of the continuous functional calculus is contained in the `elementralStarAlgebra`
+generated by the element. -/
+theorem cfc₁_mem_elementalStarAlgebra (f : C(spectrum 𝕜 a, 𝕜)) :
+    cfc₁ 𝕜 a f ∈ elementalStarAlgebra 𝕜 a :=
+  cfc₁_range_le 𝕜 a ⟨f, rfl⟩
+
+/-- The range of the continuous functional calculus is contained in the `elementalStarAlgebra`
+generated by the element. -/
+theorem cfc₂_mem_elementalStarAlgebra (f : C(𝕜, 𝕜)) : cfc₂ 𝕜 a f ∈ elementalStarAlgebra 𝕜 a :=
+  cfc₁_mem_elementalStarAlgebra _ _
+
+end Core
+
+section Normed
+
+variable (𝕜 : Type*) {A : Type*} [IsROrC 𝕜] [NormedRing A] [StarRing A] [NormedAlgebra 𝕜 A]
+    [StarModule 𝕜 A] [NormedStarGroup A] (a : A) [CompactSpace (spectrum 𝕜 a)] [CFCClass 𝕜 a]
+
+/-- For a continuous functional calculus for `a` over `IsROrC 𝕜` into a normed star ring, the
+range is precisely the `elementalStarAlgebra` generated by `a`. -/
+theorem cfc₁_range : (cfc₁ 𝕜 a).range = elementalStarAlgebra 𝕜 a := by
+  rw [StarAlgHom.range_eq_map_top, ← polynomialFunctions.starClosure_topologicalClosure, ←
+    StarSubalgebra.topologicalClosure_map _ _ (cfc₁_closedEmbedding 𝕜 a),
+    polynomialFunctions.starClosure_eq_adjoin_X, StarAlgHom.map_adjoin]
+  congr
+  rw [Set.image_singleton, toContinuousMapOnAlgHom_apply, cfc₁_map_X]
+
+-- this is the only direct result where we need the `topology.tietze_extension`
+-- and also `topology.metric_space.emetric_paracompact` for `normal_space` instance.
```

Comment:
```suggestion
-- this is the only direct result where we need the `topology.tietze_extension`
-- and also `topology.metric_space.emetric_paracompact` for `normal_space` instance.
```
This comment needs to be updated to mathlib4 terminology. I think this is talking about file imports but then there's a missing word in that sentence.